src/test/java/org/terasology/itempipes/ItemPipesTest.java

```diff
@@ -28,91 +32,108 @@
-import org.terasology.world.block.items.OnBlockItemPlaced;
 
 import java.util.EnumSet;
-import java.util.Set;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 
-public class ItemPipesTest extends ModuleTestingEnvironment {
+@ExtendWith(MTEExtension.class)
+@UseWorldGenerator("ModuleTestingEnvironment:empty")
+@Dependencies({"ItemPipes", "CoreAdvancedAssets"})
+public class ItemPipesTest {
+    @In
     private WorldProvider worldProvider;
+    @In
     private BlockManager blockManager;
+    @In
     private BlockEntityRegistry blockEntityRegistry;
+    @In
     private EntityManager entityManager;
+    @In
     private PipeSystem pipeSystem;
+    @In
     private Time time;
+    @In
+    private ModuleTestingHelper helper;
 
-    @Override
-    public Set<String> getDependencies() {
-        Set<String> modules = Sets.newHashSet();
-        modules.add("engine");
-        modules.add("CoreAdvancedAssets");
-        modules.add("SegmentedPaths");
-        modules.add("ItemPipes");
-        return modules;
-    }
+    private BlockFamily itemPipesBlockFamily;
+    private BlockFamily chestFamily;
+    private Block airBlock;
 
-    @Before
+    @BeforeEach
     public void initialize() {
-        worldProvider = getHostContext().get(WorldProvider.class);
-        blockManager = getHostContext().get(BlockManager.class);
-        blockEntityRegistry = getHostContext().get(BlockEntityRegistry.class);
-        entityManager = getHostContext().get(EntityManager.class);
-        pipeSystem = getHostContext().get(PipeSystem.class);
-        time = getHostContext().get(Time.class);
+        airBlock = blockManager.getBlock("engine:air");
+        itemPipesBlockFamily = blockManager.getBlockFamily("ItemPipes:basicPipe");
+        chestFamily = blockManager.getBlockFamily("CoreAdvancedAssets:Chest.LEFT");
+
+        BlockRegion region = new BlockRegion(0, 0, 0).expand(5, 5, 5);
+        for (Vector3ic pos : region) {
+            helper.forceAndWaitForGeneration(JomlUtil.from(pos));
+            worldProvider.setBlock(pos, airBlock);
+        }
     }
 
     @Test
     public void connectionTest() {
-        placeBlock(Vector3i.zero(), "ItemPipes:basicPipe");
-        assertEquals(0, getConn(Vector3i.zero()));
+        Vector3ic pipeLocation = new Vector3i(0, 0, 0);
+
+        placePipe(pipeLocation);
+        Assertions.assertEquals(0, getConn(new Vector3i(pipeLocation)));
 
         //begin with 1 to omit the pipe without connections.
         for (byte connections = 1; connections < 64; connections++) {
             EnumSet<Side> sideSet = SideBitFlag.getSides(connections);
 
             for (Side side : sideSet) {
-                placeBlock(side.getVector3i(), "ItemPipes:basicPipe");
+                placePipe(side.direction());
             }
 
-            assertEquals(getConn(Vector3i.zero()), connections);
+            assertEquals(getConn(new Vector3i()), connections);
```

Comment:
should this be `pipeLocation` as well?

Reply:
(That could be a reason to make the test use some initial condition that's different than the default 0-vector. If 0 is the right answer, it'd be easy for the code under test to return that _by accident_.)

Reply:
```suggestion
            assertEquals(getConn(new Vector3i(pipeLocation)), connections);
```